packages/eventual-send/src/index.d.ts

```diff
@@ -6,10 +6,12 @@ type Property = string | number | symbol;
 
 type ERef<T> = PromiseLike<T> | T;
 
+type Unpromise<T> = T extends ERef<infer U> ? U : T;
+
 // Type for an object that must only be invoked with E.  It supports a given
 // interface but declares all the functions as asyncable.
```

Comment:
The problem I'm trying to address is that an interface like `Purse` may specify some methods as synchronous.  The object I'd like to stand in for `Purse` has the same basic API, but its synchronous methods may be async instead.  I call that an `EOnly<Purse>` because it's only equivalent to a `Purse` if you only use `E` to access it.

`EOnly<T>` denotes a remote-only object, but `ERef<T>` is intended to denote a consumer that doesn't assume one of its arguments is necessarily local or remote.

The PromiseLike in `ERef<T>` is a bit of a distraction, mainly because we don't have separate types for pipelineable `Far`s (which have useful identities, but remote-only methods) vs. eventual data that can meaningfully be awaited/thened.  We'll probably have to clean this up if we support local properties on remote objects.